### PR TITLE
Prepare migration

### PR DESCRIPTION
**Pull request by benoitdevos**
### Prepare migration
* Configure Logion pallets in runtime (Re-use weights calculated for solo).
* Adapt genesis config.
* Remove useless migration.

logion-network/logion-internal#1199

### CODE REVIEW

File: runtime/src/lib.rs [hunk 1]
```diff
@@ -167,7 +167,6 @@ impl<T: Config> OnRuntimeUpgrade for ClearHostConfiguration<T> {
 #[allow(unused_parens)]
 type Migrations = (
 	ClearHostConfiguration<Runtime>,
```
Comment (gdethier) [inline runtime/src/lib.rs]: To answer the question in this PR's description: no, this migration is not needed anymore and may safely be removed.

File: runtime/src/lib.rs [hunk 2]
```diff
@@ -212,7 +211,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	spec_name: create_runtime_str!("logion"),
 	impl_name: create_runtime_str!("logion"),
 	authoring_version: 1,
-	spec_version: 3,
+	spec_version: 4,
```
Comment (gdethier) [inline runtime/src/lib.rs]: Shall we apply semver spec_versions? If yes, spec version should be 3000 (for 0.3.0 -> 000_003_000 i.e. 3000).

File: runtime/src/lib.rs [hunk 3]
```diff
@@ -308,23 +307,51 @@ parameter_types! {
...
 #[derive_impl(frame_system::config_preludes::ParaChainDefaultConfig as frame_system::DefaultConfig)]
 impl frame_system::Config for Runtime {
+	/// The basic call filter to use in dispatchable.
+	type BaseCallFilter = BaseCallFilter;
 	/// The identifier used to distinguish between accounts.
 	type AccountId = AccountId;
+	/// The aggregated dispatch type that is available for extrinsics.
+	type RuntimeCall = RuntimeCall;
+	/// The lookup mechanism to get account ID from whatever is passed in dispatchers.
+	type Lookup = AccountIdLookup<AccountId, ()>;
```
Comment (gdethier) [inline runtime/src/lib.rs]: I think that `RuntimeCall` and `Lookup` are automatically set by `derive_impl`.

File: runtime/src/lib.rs [hunk 4]
```diff
@@ -308,23 +307,51 @@ parameter_types! {
...
 	/// The type for hashing blocks and tries.
 	type Hash = Hash;
+	/// The hashing algorithm used.
+	type Hashing = BlakeTwo256;
 	/// The block type.
 	type Block = Block;
+	/// The ubiquitous event type.
+	type RuntimeEvent = RuntimeEvent;
+	/// The ubiquitous origin type.
+	type RuntimeOrigin = RuntimeOrigin;
```
Comment (gdethier) [inline runtime/src/lib.rs]: Same remark.

File: runtime/src/lib.rs [hunk 5]
```diff
@@ -308,23 +307,51 @@ parameter_types! {
...
+	/// The aggregated dispatch type that is available for extrinsics.
+	type RuntimeCall = RuntimeCall;
+	/// The lookup mechanism to get account ID from whatever is passed in dispatchers.
+	type Lookup = AccountIdLookup<AccountId, ()>;
 	/// The index type for storing how many extrinsics an account has signed.
 	type Nonce = Nonce;
 	/// The type for hashing blocks and tries.
 	type Hash = Hash;
+	/// The hashing algorithm used.
+	type Hashing = BlakeTwo256;
```
Comment (gdethier) [inline runtime/src/lib.rs]: Same remark.

File: runtime/src/lib.rs [hunk 6]
```diff
@@ -308,23 +307,51 @@ parameter_types! {
...
+	/// The ubiquitous origin type.
+	type RuntimeOrigin = RuntimeOrigin;
 	/// Maximum number of block number to block hash mappings to keep (oldest pruned first).
 	type BlockHashCount = BlockHashCount;
 	/// Runtime version.
 	type Version = Version;
+	/// Converts a module to the index of the module in `construct_runtime!`.
+	///
+	/// This type is being generated by `construct_runtime!`.
+	type PalletInfo = PalletInfo;
```
Comment (gdethier) [inline runtime/src/lib.rs]: Same remark.

File: runtime/src/lib.rs [hunk 7]
```diff
@@ -360,20 +389,87 @@ parameter_types! {
...
 	type MaxLocks = ConstU32<50>;
+	type MaxReserves = ();
+	type ReserveIdentifier = [u8; 8];
 	/// The type for recording an account's balance.
 	type Balance = Balance;
 	/// The ubiquitous event type.
 	type RuntimeEvent = RuntimeEvent;
 	type DustRemoval = ();
-	type ExistentialDeposit = ExistentialDeposit;
+	type ExistentialDeposit = ConstU128<EXISTENTIAL_DEPOSIT>;
```
Comment (gdethier) [inline runtime/src/lib.rs]: Either re-use `ExistentialDeposit` parameter, either remove it.

File: runtime/src/lib.rs [hunk 8]
```diff
@@ -360,20 +389,87 @@ parameter_types! {
...
 	type AccountStore = System;
-	type WeightInfo = pallet_balances::weights::SubstrateWeight<Runtime>;
-	type MaxReserves = ConstU32<50>;
-	type ReserveIdentifier = [u8; 8];
-	type RuntimeHoldReason = RuntimeHoldReason;
-	type RuntimeFreezeReason = RuntimeFreezeReason;
```
Comment (gdethier) [inline runtime/src/lib.rs:374]: Parameters `RuntimeHoldReason` and `RuntimeFreezeReason` should be re-used or removed.

File: runtime/src/lib.rs [hunk 9]
```diff
@@ -360,20 +389,87 @@ parameter_types! {
...
+    pub const InclusionFeesDistributionKey: DistributionKey = DistributionKey {
+        legal_officers_percent: Percent::from_percent(35),
+        community_treasury_percent: Percent::from_percent(30),
+        logion_treasury_percent: Percent::from_percent(35),
+        loc_owner_percent: Percent::from_percent(0),
+    };
+
+	// Inflation: I=0,05 (5%)
+	// Total supply: N=10^9
+	// Block rate: B=6 (Number of seconds between 2 blocks)
```
Comment (gdethier) [inline runtime/src/lib.rs]: Block rate is 12 on the parachain so `InflationAmount` should be adjusted to take this into account.

File: runtime/src/lib.rs [hunk 10]
```diff
@@ -382,24 +478,32 @@ pub struct DealWithInclusionFees;
...
 }
 
 pub type WeightToFee = ConstantMultiplier<Balance, WeightToFeeMultiplier>;
 
 parameter_types! {
 	pub FeeMultiplier: Multiplier = Multiplier::one();
-	pub const WeightToFeeMultiplier: Balance = 10_000_000;
+
+	// The multiplier is set such as inclusion fees are ~2 LGNT on average.
+	// Spreadsheet in /docs/inclusion_fees.ods contains the model that lead
```
Comment (gdethier) [inline runtime/src/lib.rs:450]: The spreadsheet should be added to collator's code base.

File: runtime/src/lib.rs [hunk 11]
```diff
@@ -532,7 +636,315 @@ impl pallet_collator_selection::Config for Runtime {
...
+	type TokensRecordIdFactory = ();
+	#[cfg(feature = "runtime-benchmarks")]
+	type EthereumAddressFactory = ();
+	#[cfg(feature = "runtime-benchmarks")]
+	type SponsorshipIdFactory = ();
+}
+
+parameter_types! {
+	pub const RecoveryConfigDepositBase: u64 = 10;
+	pub const RecoveryFrieldDepositFactor: u64 = 1;
```
Comment (gdethier) [inline runtime/src/lib.rs]:
```suggestion
	pub const RecoveryFriendDepositFactor: u64 = 1;
```

File: runtime/src/lib.rs [hunk 12]
```diff
@@ -532,7 +636,315 @@ impl pallet_collator_selection::Config for Runtime {
...
+	pub const MaxFriends: u16 = 3;
+	pub const RecoveryDeposit: u64 = 10;
+}
+
+impl pallet_recovery::Config for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	type RuntimeCall = RuntimeCall;
+	type Currency = Balances;
+	type ConfigDepositBase = RecoveryConfigDepositBase;
+	type FriendDepositFactor = RecoveryFrieldDepositFactor;
```
Comment (gdethier) [inline runtime/src/lib.rs]:
```suggestion
	type FriendDepositFactor = RecoveryFriendDepositFactor;
```

**Review by gdethier — CHANGES_REQUESTED** [review]
Good job, I have a few comments that IMHO must be handled before the branch can be merged.

File: runtime/src/lib.rs (hunk 1 above)
Comment (benoitdevos) [reply runtime/src/lib.rs]: Removed

File: runtime/src/lib.rs (hunk 3 above)
Comment (benoitdevos) [reply runtime/src/lib.rs]: Indeed.

**Review by gdethier — APPROVED** [review]
:+1:

**Pull request merged**